src/scheduler/data_types.h

```diff
@@ -1126,54 +1133,50 @@ struct config
 	bool non_prime_pre:1;
 	bool update_comments:1;	/* should we update comments or not */
 	bool prime_exempt_anytime_queues:1; /* backfill affects anytime queues */
-	bool assign_ssinodes:1;	/* assign the ssinodes resource */
 	bool preempt_starving:1;	/* once jobs become starving, it can preempt */
 	bool preempt_fairshare:1; /* normal jobs can preempt over usage jobs */
 	bool dont_preempt_starving:1; /* don't preempt staving jobs */
 	bool enforce_no_shares:1;	/* jobs with 0 shares don't run */
 	bool node_sort_unused:1;	/* node sorting by unused/assigned is used */
-	bool resv_conf_ignore:1;  /* if we want to ignore dedicated time when confirming reservations.  Move to enum if ever expanded */
-	bool allow_aoe_calendar:1;        /* allow jobs requesting aoe in calendar*/
+	bool resv_conf_ignore:1;	/* if we want to ignore dedicated time when confirming reservations.  Move to enum if ever expanded */
+	bool allow_aoe_calendar:1;	/* allow jobs requesting aoe in calendar*/
 #ifdef NAS /* localmod 034 */
-	bool prime_sto	:1;	/* shares_track_only--no enforce shares */
+	bool prime_sto:1;	/* shares_track_only--no enforce shares */
 	bool non_prime_sto:1;
 #endif /* localmod 034 */
 
-	struct sort_info *prime_sort;		/* prime time sort */
-	struct sort_info *non_prime_sort;	/* non-prime time sort */
+	std::vector<sort_info> prime_sort;	/* prime time sort */
+	std::vector<sort_info> non_prime_sort;	/* non-prime time sort */
 
 	enum smp_cluster_dist prime_smp_dist;	/* how to dist jobs during prime*/
 	enum smp_cluster_dist non_prime_smp_dist;/* how do dist jobs during nonprime*/
 	time_t prime_spill;			/* the amount of time a job can
 						 * spill into primetime
 						 */
 	time_t nonprime_spill;			/* vice versa for prime_spill */
-	fairshare_head *fairshare;		/* fairshare tree */
 	time_t decay_time;			/*  time in seconds for the decay period*/
-	struct t prime[HIGH_DAY][2];	/* prime time start and prime time end*/
-	int holidays[MAX_HOLIDAY_SIZE];		/* holidays in Julian date */
+	struct t prime[HIGH_DAY][2];		/* prime time start and prime time end*/
```

Comment:
for future, i think we should start using std::array for static arrays, it provides similar advantages as std::string vs char array, and comes with a useful set of built-in functions.